client/httpClient.go

```diff
@@ -0,0 +1,143 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/node-monitoring/common"
+	"github.com/ElrondNetwork/node-monitoring/config"
+	"github.com/ElrondNetwork/node-monitoring/data"
+)
+
+// TODO: refactor to separate http client wrapper from event logic
+
+var log = logger.GetOrCreate("client/httpClient")
+
+const (
+	firstRunValue float64 = -1
+	maxRating             = 10000000
+)
+
+const (
+	baseURL = "https://api.elrond.com"
+
+	nodesIdentifierPath = "/nodes?identity=%s"
+)
+
+type httpClientWrapper struct {
+	httpClient    *http.Client
+	mutHTTPClient sync.RWMutex
+	lastValue     float64
+	config        *config.NodeRating
+}
+
+// HTTPClientWrapperArgs defines the arguments needed to create a new client
+type HTTPClientWrapperArgs struct {
+	ReqTimeoutSec int
+	Config        *config.NodeRating
+}
+
+// NewHTTPClientWrapper creates an instance of httpClient which is a wrapper for http.Client
+func NewHTTPClientWrapper(args HTTPClientWrapperArgs) (*httpClientWrapper, error) {
+	httpClient := http.DefaultClient
+	httpClient.Timeout = time.Duration(args.ReqTimeoutSec) * time.Second
+
+	return &httpClientWrapper{
+		httpClient: httpClient,
+		lastValue:  firstRunValue,
+		config:     args.Config,
+	}, nil
+}
+
+func (hcw *httpClientWrapper) GetEvent() (data.NotificationMessage, error) {
+	event := data.NotificationMessage{}
+
+	identity := hcw.config.Identity
+	percentageThreshold := hcw.config.Threshold
+
+	if hcw.lastValue == firstRunValue {
+		return event, nil
+	}
+
+	path := fmt.Sprintf(nodesIdentifierPath, identity)
+	nodes, err := hcw.callGetRestEndPoint(baseURL, path)
+	if err != nil {
+		return event, err
+	}
+
+	msg := ""
+	for _, node := range nodes {
+		diff := math.Abs(node.TempRating - hcw.lastValue)
+		changePercentage := diff / float64(maxRating)
+		if changePercentage > percentageThreshold {
```

Comment:
maybe add a TODO to change this behavior in a future task:
as discussed, we might avoid triggering the alarm if the rating drops under the set threshold. 
Can be left as it is for now, we will test this with the threshold set to 1%

Reply:
added TODO